### PR TITLE
:editor parinfer for lisp editing

### PR DESCRIPTION
**Pull request by ar1a**

### CODE REVIEW

File: modules/editor/parinfer/config.el [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    (parinfer-mode))
+  (setq parinfer-extensions
+        '(defaults
+           pretty-parens
+           smart-tab
+           smart-yank))
+  (if (featurep! :feature evil)
+      (push 'evil parinfer-extensions))
+  :config
+  (map! :map emacs-lisp-mode-map
```
Comment (mattly) [inline modules/editor/parinfer/config.el]: In my own config segment that this will replace, I've mapped these under `parinfer-mode-map`.

**Comment by hlissner** [comment]
I welcome this PR, but please rebase that `fixup!`

File: modules/editor/parinfer/config.el (hunk 1 above)
Comment (ar1a) [reply modules/editor/parinfer/config.el]: great idea

**Comment by ar1a** [comment]
done!

**Comment by hlissner** [comment]
Excellent work, thanks again!

**Pull request merged**